test/e2e/util_test.go

```diff
@@ -649,6 +673,20 @@ func updateInfrastructureConfigStatusWithRetryOnConflict(t *testing.T, timeout t
 	})
 }
 
+// createWithRetryOnConflict creates the given object. If there is a conflict error on create
+// then the create is retried until the timeout is reached.
+func createWithRetryOnConflict[T client.Object](ctx context.Context, obj T, timeout time.Duration) error {
```

Comment:
I don't think you need generics/type parameters here, you can just pass client.Object as the argument of object. kclient already accepts a client.Object